picard/file.py

```diff
@@ -442,6 +442,7 @@ def _move_additional_files(self, old_filename, new_filename):
                 file_names = filtered_names
             for old_file in file_names:
                 if pattern_regex.match(old_file):
+                    file_names.remove(old_file)
```

Comment:
That makes the loop O(n2). Can we make file_names a set instead so that it's O(n)?